src/main/java/org/humancellatlas/ingest/notifications/NotificationConfiguration.java

```diff
@@ -0,0 +1,86 @@
+package org.humancellatlas.ingest.notifications;
+
+import java.util.Collection;
+import java.util.Collections;
+import lombok.Getter;
+import org.humancellatlas.ingest.messaging.Constants.Exchanges;
+import org.humancellatlas.ingest.messaging.Constants.Routing;
+import org.humancellatlas.ingest.notifications.processors.NotificationProcessor;
+import org.humancellatlas.ingest.notifications.processors.impl.email.EmailNotificationProcessor;
+import org.humancellatlas.ingest.notifications.processors.impl.email.SMTPConfig;
+import org.humancellatlas.ingest.notifications.sources.NotificationSource;
+import org.humancellatlas.ingest.notifications.sources.impl.inmemory.InmemoryNotificationSource;
+import org.humancellatlas.ingest.notifications.sources.impl.rabbit.AmqpConfig;
+import org.humancellatlas.ingest.notifications.sources.impl.rabbit.RabbitNotificationSource;
+import org.springframework.amqp.rabbit.core.RabbitMessagingTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+@Configuration
+public class NotificationConfiguration {
+
+  @Component
+  @Getter
+  class SmtpEnvVars {
+    private final String smtpHost;
+    private final int smtpPort;
+    private final String username;
+    private final String password;
+
+    @Autowired
+    public SmtpEnvVars(Environment environment) {
+      this.smtpHost = environment.getProperty("NOTIFICATIONS_SMTP_HOST", "localhost");
+      this.smtpPort = Integer.parseInt(environment.getProperty("NOTIFICATIONS_SMTP_PORT", "587"));
+      this.username = environment.getProperty("NOTIFICATIONS_SMTP_USERNAME", "admin");
+      this.password = environment.getProperty("NOTIFICATIONS_SMTP_PASSWORD", "password");
+    }
+  }
+
+  @Component
+  @Getter
+  class AmqpEnvVars {
+    private final String sendExchange;
+    private final String sendRoutingKey;
+
+    @Autowired
+    public AmqpEnvVars(Environment environment) {
+      this.sendExchange = environment.getProperty("NOTIFICATIONS_AMQP_SEND_EXCHANGE",
+                                                  Exchanges.NOTIFICATIONS_EXCHANGE);
+      this.sendRoutingKey = environment.getProperty("NOTIFICATIONS_AMQP_SEND_ROUTING_KEY",
+                                                    Routing.ADD_NOTIFICATION);
+    }
+  }
+
+  @Bean
+  public SMTPConfig smtpConfig(SmtpEnvVars smtpEnvVars) {
```

Comment:
This is to isolate configuration (SMTPConfig) from an IO source of configuration properties (SmtpEnvVars). Configuration shouldn't necessarily depend on any particular IO.

Reply:
As noted in the other comment, you can instead bind `SMTPConfig` to application properties and let Spring take care of where to source it.

Reply:
Spring configuration looks like a good call for the SmtpEnvVars (will rename it to something else). The SMTPConfig class on the other hand is intended to hold the many parameters required to construct the email sender (could unroll all those parameters and use a builder instead). I personally wouldn't make it dependent on spring configuration since it's purpose is to represent a typed dictionary of params, in essence.